cmd/simulacra/simulacra.go

```diff
@@ -178,43 +154,100 @@ func configureFromGCloud(project *string, zone *string) error {
-	thirdPartyAppsPath := flag.String("install_path", "./integration_test/third_party_apps_data", "Optional. The path to the third party apps data folder. If missing, Simulacra assumes the working directory is the root of the repo. Therefore, the default path is './integration_test/third_party_apps_data' ")
+	name := flag.String("name", getInstanceName(), "Optional. A name for the instance to be created. If missing, a random name with prefix 'simulacra-vm-instance' will be assigned. ")
+	thirdPartyAppsPath := flag.String("install_path", defaultThirdPartyAppsPath, "Optional. The path to the third party apps data folder. If missing, Simulacra assumes the working directory is the root of the repo. Therefore, the default path is './integration_test/third_party_apps_data' ")
 	flag.Parse()
 
-	if err := configureFromGCloud(project, zone); err != nil {
-		logger.Fatalf("project and zone must either be non empty or set in GCloud %v", err)
+	if *configPath != "" {
+		return getConfigFromYaml(*configPath)
+	}
+
+	config := Config{Platform: *platform, ConfigFilePath: *opsAgentConfigFile, Project: *project, Zone: *zone, Name: *name,
+		ThirdPartyAppsPath: *thirdPartyAppsPath}
+
+	return config, nil
+
+}
+
+func main() {
+	loggingDir := fmt.Sprintf("/tmp/simulacra-%s", uuid.NewString())
+	logger, err := logging.NewDirectoryLogger(loggingDir)
+	if err != nil {
+		log.Default().Fatalf("Error initializing directory logger %v", err)
+	}
+	log.Default().Printf("Starting Simulacra, Detailed logging can be found in %s directory", loggingDir)
+	ctx := context.Background()
+	config, err := getSimulacraConfig()
+	if err != nil {
+		log.Default().Fatalf("error parsing simulacra config %v", err)
+	}
+
+	if err := configureFromGCloud(&config.Project, &config.Zone); err != nil {
+		log.Default().Fatalf("project and zone must either be non empty or set in GCloud %v", err)
 	}
 
 	options := gce.VMOptions{
-		Platform:    *platform,
-		MachineType: agents.RecommendedMachineType(*platform),
-		Name:        *name,
-		Project:     *project,
-		Zone:        *zone,
+		Platform:    config.Platform,
+		MachineType: agents.RecommendedMachineType(config.Platform),
+		Name:        config.Name,
+		Project:     config.Project,
+		Zone:        config.Zone,
 	}
 	// Create VM Instance.
-	vm, err := gce.CreateInstance(ctx, logger, options)
+	log.Default().Printf("Creating VM Instance, check %s for details", vmInitLogFile)
+	vm, err := gce.CreateInstance(ctx, logger.ToFile(vmInitLogFile), options)
 	if err != nil {
-		logger.Fatalf("Failed to create GCE instance %v", err)
+		log.Default().Fatalf("Failed to create GCE instance %v", err)
 	}
 	// Install Ops Agent on VM.
-	if err := setupOpsAgent(ctx, vm, logger, *configFile); err != nil {
-		logger.Fatalf("Failed to install Ops Agent %v", err)
+	log.Default().Print("Installing Ops Agent, check main_log.txt for details")
+	if err := setupOpsAgent(ctx, vm, logger.ToMainLog(), config.ConfigFilePath); err != nil {
```

Comment:
We can put logs for Ops Agent installation in a separate file but because there's not much that usually arises i figured it's best to just keep it to main_log.txt